### ISSUE TITLE
YCBA objects loaned to exhibitions

### ISSUE THREAD

**Issue by edgartdata** [ycba-cia/blacklight-collections2#397]
@Lcallery, all, there is interest in having the status of YCBA objects loaned to exhibitions in their online records. This might have to be vetted with the registrars who in the past have been cautious about sharing information about our object in exhibitions until they come back to the YCBA. There may be security issues around this topic.
If it were possible, would we want to share which exhibitions the objects are in at MFA Boston or Paris for example?

**Comment by edgartdata** [comment]
@yulgit1 I have created this new TMS object package _**ycba:loans to YUAG In a New Light 2023 exhibition**_ with 56 objects to test how their new access status would come through in BL. Could we say "On view at the Yale University Art Gallery" instead of "Not on View" for these 56 objects?

**Comment by Lcallery** [comment]
The RBM and P&D items will not be accessible to the Study Room while out of the building (which I think they are already).   Should this change be noted somehow as well?

**Comment by edgartdata** [comment]
@Lcallery Yes, and Communications is working on language for that. Jackie will let us know soon (see https://github.com/ycba-cia/blacklight-collections2/issues/436). 

Issue 397 is about YCBA objects loaned to exhibitions, and specifically to the YUAG In a New Light exhibition while the museum is closed. From TMS, there are no RBM or P&D objects loaned to the YUAG show.

**Comment by Lcallery** [comment]
My mistake.  Sorry

**Comment by edgartdata** [comment]
No worries. It's all connected.

**Comment by yulgit1** [comment]
The 56 loan objects should have "On view at the Yale University Art Gallery" in harvester-test LIDO tomorrow.

https://github.com/ycba-cia/coboat-github/commit/7a62759cd0913e2562521f19d1e72b064fb6dd69

https://harvester-test.britishart.yale.edu/oaicatmuseum/OAIHandler?verb=GetRecord&identifier=oai:tms.ycba.yale.edu:34&metadataPrefix=lido

34
45
205
324
406
488
562
567
568
625
635
644
646
648
658
667
870
896
1004
1082
1095
1099
1227
1287
1381
1403
1566
4995
5000
5001
5008
5009
5010
5011
5016
5018
5021
5023
5024
5026
5027
5033
5036
5046
5048
5052
5059
5062
5065
38488
38526
38536
38543
67320
73253
74213

**Comment by edgartdata** [comment]
@yulgit1 34 and 45 don't have the new access status today 2/9 :(

**Comment by yulgit1** [comment]
Yes, one example, last nights run doesn't have it:

`73253	B2016.24	73253	1	On view`

But when I run it now it does:

`73253	B2016.24	73253	1	On view at the Yale University Art Gallery`

Kind of defies explanation, will see tomorrow.

**Comment by edgartdata** [comment]
It ran during the weekend and the new access note for these objects is in test LIDO XML.

Important note: the exhibition "In a New Light" opens at YUAG on Friday, March 24, 2023 and closes on Sunday, December 3, 2023.

I will check with Communications when they want the new access status to go up in BlackLight.

**Comment by yulgit1** [comment]
Note to self.  Will need to modify:
https://git.yale.edu/ermadmix/camel_collections2/blob/master/src/main/java/edu/yale/britishart/camel/workflow/ProcessHelper.java#L181

**Comment by edgartdata** [comment]
@yulgit1 On February 27 can we please change the Access note for the YCBA objects in the YUAG "In a New Light" exhibition to ready:
“Will be on view at the Yale University Art Gallery beginning March 24”

On March 24, once the YUAG exhibition opens, we will need to change it to read:
“On view at the Yale University Art Gallery”

**Comment by yulgit1** [comment]
Yes, an indicator in the MARC.

**Comment by flapka** [comment]
We normally record exhibition activity in MARC field 711. If we add that field for all objects in Crafting Worldviews, could we than apply a rule that says when 711 subfield $0 = "(CtY-BA)1072" access is at YUAG (until June 25), otherwise (for everything else) access is the RBM default.

**Comment by edgartdata** [comment]
@yulgit1 Eric, I have just added the following art objects to the TMS object package **_ycba:loans to YUAG In a New Light 2023 exhibition_** (I realize that the name of the package is misleading but I did not want to rename it in case it would be problematic for you):

The 3 YCBA paintings that are at YUAG but adjacent to the "In a New Light" exhibition:
417
5064
4988

The 4 YCBA works that are at YUAG in the "Crafting WorldViews" exhibition:
41888
855
5042
1176

So **_ycba:loans to YUAG In a New Light 2023 exhibition_** now contains 63 objects.

Let's plan to change their access status in the item records and in the On-Site Access to "On view at the Yale University Art Gallery" on March 23 so that they are updated online on the 24th. I know that the 4 "Crafting Worldviews" are already at YUAG now but that's too complicated.

**Comment by yulgit1** [comment]
In a New Light 2023 locations for the 63 to index tonight.

https://github.com/ycba-cia/coboat-github/commit/d2dccd8dc8b7111efc7516b9f70b012acfaf156d
https://git.yale.edu/ermadmix/camel_collections2/commit/3cc7a8fb16554b3400d256a092912302cef87b6c

**Comment by edgartdata** [comment]
For RBM, returning to @Lcallery's suggestion earlier, RBM has eight items in Crafting worldviews: https://artgallery.yale.edu/exhibitions/exhibition/crafting-worldviews-art-and-science-europe-1500-1800 

@flapka @yulgit1 It would be helpful if they also got the access note/facet value of "On View at the Yale University Art Gallery".

**Comment by yulgit1** [comment]
AFAICT, The facet would have to come from the MARC, and the Holdings from the mfhd like:

https://libapp.library.yale.edu/OAI_BAC/src/OAIOrbisTool.jsp?verb=GetRecord&identifier=oai:orbis.library.yale.edu:739403&metadataPrefix=marc21

https://libapp.library.yale.edu/VoySearch/GetAllMfhdItem?bibid=739403

**Comment by flapka** [comment]
@yulgit1 That's correct, it would be best for the data to be in the Holdings. But if it were there, could it be indexed? I thought we weren't indexing any Holdings data (because of performance issues).

On a related note, I probably have the capacity now to renew exploration of harvesting from https://pod.stanford.edu/, if the same goes for you.

**Comment by yulgit1** [comment]
Damn, right.  The facets part require indexing.  I'm at full capacity right now, but will get at PODS when I can.

**Comment by flapka** [comment]
@yulgit1 If we instead use data in the MARC bib record, for now at least, could we apply a logic as follows?

If 711 subfield $0 = "(CtY-BA)1072" [i.e. Crafting world views] then access = "On view at the Yale University Art Gallery"

otherwise default to the default bacrb access value "Accessible by appointment in the Study Room"

**Comment by yulgit1** [comment]
@flapka That should work technically.  Not sure I understand what 1072 indicates and why this is "ie Crafting world views."

Anyway, could you give an example record?

**Comment by flapka** [comment]
See example record: https://libapp.library.yale.edu/OAI_BAC/src/OAIOrbisTool.jsp?verb=GetRecord&identifier=oai:orbis.library.yale.edu:11416653&metadataPrefix=marc21

The 711 field is as follows:
```
<marc:datafield tag="711" ind1="2" ind2=" ">
<marc:subfield code="i">Exhibited in:</marc:subfield>
<marc:subfield code="a">Crafting Worldviews: European Instruments of Art and Science from Yale University</marc:subfield>
<marc:subfield code="c">(Yale University Art Gallery,</marc:subfield>
<marc:subfield code="d">February 17, 2023-June 24, 2023)</marc:subfield>
<marc:subfield code="0">(CtY-BA)1072</marc:subfield>
<marc:subfield code="5">CtY-BA</marc:subfield>
</marc:datafield>
```

**Comment by edgartdata** [comment]
Related question for @flapka: is there a way to derive content for an Exhibition History tab for RBM item records from MARC?

**Comment by yulgit1** [comment]
@flapka just a heads up indexing is off this week as we are running a TMAN DB migration

**Comment by edgartdata** [comment]
And i have just asked Nancy to set the status of the Crafting Worldviews exhibition to completed so that YCBA art and RBM objects that are in it get their access note/facet and exhibition history updated. Sorry I did not ask her before.

**Comment by yulgit1** [comment]
Change in the indexer to pick up 711 subfield 0 with 1072. To run overnight:

https://git.yale.edu/ermadmix/ycba_xslts/commit/e97a4df6c0423f85c7636a665fcc87a8c498c8f4
https://git.yale.edu/ermadmix/camel_collections2/commit/f5fe0d2d154a1b4f3e535e4ef19d5cb4941f9e23

**Comment by flapka** [comment]
Great. In the access facet, the 8 in RBM now show up with the correct YUAG value. But in the detailed object view, they still show up with the RBM default (Study Room). How hard would it be to update the mapping there too?

**Comment by edgartdata** [comment]
Just noting here that Francis' suggestion just above has been addressed. In other words, for RBM that are on view at YUAG, the detailed object view do not have "View by request in the Study Room" anymore.

**Comment by yulgit1** [comment]
This issue also tracked at:
https://github.com/ycba-cia/blacklight-collections2/issues/454

**Comment by flapka** [comment]
@yulgit1 @edgartdata The Crafting Worldviews exhibition at YUAG ends on Sunday (June 25). The RBM access values for the eight objects in the exhibition were made possible by a temporary change to the indexing (based on a value in field 711, as described above). Can we revert the indexing next week, so these RBM items will no longer appear as on view?

**Comment by yulgit1** [comment]
Yes I will revert next week:

https://git.yale.edu/ermadmix/ycba_xslts/commit/e97a4df6c0423f85c7636a665fcc87a8c498c8f4

One thing though, I can not access the spinup server the indexer is on since I got back yesterday.  It just hangs trying to make the ssh connection.  I put in a ticket with ITS, hopefully they will respond soon.

**Comment by yulgit1** [comment]
Reverted.

Note, something caused the indexing to fail.  Root cause on the spinup indexing server:

Caused by: java.lang.OutOfMemoryError: unable to create new native thread

I believe this caused a cascade of errors, that corrupted the solr index and made blacklight unreachable.  I was finally able to recover the server.  Crons are all set to run tonight. Will see tomorrow.

**Comment by yulgit1** [comment]
Server spinup-00201a.spinup.yale.edu was recovered with a stop and start

indexing has ran successfully past 3 nights
